state/core_access.go

```diff
@@ -224,6 +224,19 @@ func (ca *CoreAccessor) SubmitPayForBlob(
 	fee Int,
 	gasLim uint64,
 	blobs []*blob.Blob,
+) (*TxResponse, error) {
+	return ca.SubmitPayForBlobWithOptions(ctx, fee, gasLim, blobs, nil)
+}
+
+// SubmitPayForBlobWithOptions builds, signs, and submits a MsgPayForBlob with an additional options defined
+// in `PayForBlobOptions`. If gasLim is set to 0, the method will automatically estimate the gas limit.
+// If the fee is negative, the method will use the nodes min gas price multiplied by the gas limit.
+func (ca *CoreAccessor) SubmitPayForBlobWithOptions(
+	ctx context.Context,
+	fee Int,
+	gasLim uint64,
```

Comment:
[discussion] - should gasLim be added as an option bc it's not necessary to set it? if the field is not set in options, it would just do the calculation similar to if a negative fee is passed.

cc @Wondertan

Reply:
@cmwaters

Reply:
@liamsi see the comment above regarding
> Having an options struct part of the JSON-RPC further more manifests that this current API is basically driven by implementation details

Reply:
Can we please first add the option to define which account to use for signing before adding more things to the options? That said, if gasLim is optional, yes it should go into options. Maybe we can remove it from this Pr instead though. Introduce the only important option first and then bikeshed about other options are to be added?

Reply:
My suggestion is to be iterative and to do everything step by step. I mean, let's merge this pr with empty options and then we can extend the struct whenever we need.

Reply:
This has an issue making the main branch unreleasable. I think we should stick to an approach where we implement the whole feature fully and merge it in the main. We can either do this through the feature branch or through a single PR that has the whole feature

Reply:
Feature branch also works for me

Reply:
Currently, gasLim is passed as a parameter, which would prevent moving it to options later without breaking the API. To avoid this, let remove gasLim from the parameters, allowing for optional future configuration.

Reply:
Let's consolidate all parameters that can default to implicit values into the options structure in this PR.